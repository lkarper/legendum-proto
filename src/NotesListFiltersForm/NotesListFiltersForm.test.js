import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import NotesListFiltersForm from './NotesListFiltersForm';

describe('NotesListFiltersForm component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<NotesListFiltersForm />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<NotesListFiltersForm />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
