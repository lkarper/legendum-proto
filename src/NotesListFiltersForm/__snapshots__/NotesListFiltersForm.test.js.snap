// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NotesListFiltersForm component renders the UI as expected 1`] = `
<NotesListFiltersForm
  chapterFilter="all"
  setChapterFilter={[Function]}
  setSortType={[Function]}
  sortType="rec"
  suffix=""
>
  <form
    className="NotesListFiltersForm__form-"
  >
    <fieldset>
      <legend>
        Sort by date saved
      </legend>
      <div>
        <input
          checked={true}
          className="NotesListFiltersForm__input radio"
          id="sort-rec"
          name="sort-notes"
          onChange={[Function]}
          type="radio"
          value="rec"
        />
        <label
          htmlFor="sort-rec"
        >
          Sort notes newest - oldest
        </label>
      </div>
      <div>
        <input
          checked={false}
          className="NotesListFiltersForm__input radio"
          id="sort-old"
          name="sort-notes"
          onChange={[Function]}
          type="radio"
          value="old"
        />
        <label
          htmlFor="sort-old"
        >
          Sort notes oldest - newest
        </label>
      </div>
    </fieldset>
  </form>
</NotesListFiltersForm>
`;
